.github/workflows/docker.yml

```diff
@@ -0,0 +1,53 @@
+name: Docker
+
+on:
+  push:
+    branches:
+      - 'bahdotsh/test-workflow'
+  release:
+    types: [published]
+
+
+jobs:
+  docker:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: name/persistencecore
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: checkout repo
+        uses: actions/checkout@v2
+
+      - name: build docker-image
+        if: github.event.push
+        run: make docker-build PROCESS=persistencecore DOCKER_IMAGE_NAME=persistenceone/persistencecore DOCKER_TAG_NAME=latest
+
+      - name: push docker-image
+        if: github.event.push
+        run: make docker-push IMAGE_NAME=persistenceone/persistencecore TAG_NAME=latest
```

Comment:
we need to pushing the tag name as well and not just latest

Reply:
If it's from a release we are using the tag name as well

Reply:
3 tags as we discussed

Reply:
you need to have a step that determines the tag, based on what the github context was this should give the tag with the commit or the release itself.
for latest, is just a condition if there was a push to the main